ddtrace/internal/forksafe.py

```diff
@@ -3,39 +3,41 @@
 """
 import logging
 import os
-
-
-__all__ = [
-    "ddtrace_after_in_child",
-    "register",
-]
+import typing
 
 
 log = logging.getLogger(__name__)
 
 
-registry = []
+_registry = []
 
 
 def ddtrace_after_in_child():
-    global registry
+    # type: () -> None
+    global _registry
 
-    for hook in registry:
+    for hook in _registry:
         try:
             hook()
         except Exception:
             # Mimic the behaviour of Python's fork hooks.
             log.exception("Exception ignored in forksafe hook %r", hook)
 
 
-def _register(hook):
-    if hook not in registry:
-        registry.append(hook)
+def register(after_in_child):
```

Comment:
I wonder if it would make sense to make the registry a `set` and check that one is not registering the same function multiple times by accident

Reply:
I don't think so. Also order can matters.

Reply:
On the point of ordering, IMHO using `register` as a decorator won't enforce any particular ordering, unless decorated functions/methods are also moved around and import order is also enforced. Simply decorating existing functions will result in things being registered in random order

Reply:
That's why order *can* matter.